spark/v3.2/spark/src/main/scala/org/apache/spark/sql/execution/datasources/SparkExpressionConverter.scala

```diff
@@ -30,4 +34,18 @@ object SparkExpressionConverter {
     // But these two conversions already exist and well tested. So, we are going with this approach.
     SparkFilters.convert(DataSourceStrategy.translateFilter(sparkExpression, supportNestedPredicatePushdown = true).get)
   }
+
+  @throws[AnalysisException]
+  def collectResolvedSparkExpression(session: SparkSession, tableName: String, where: String): Expression = {
+    var expression:Expression = null
+    // Add a dummy prefix linking to the table to collect the resolved spark expression from analyzed plan.
+    val prefix = String.format("SELECT 42 from %s where ", tableName)
```

Comment:
nit: L42-43 can be combined, `prefix` is never used after L43.

Reply:
I kept it intentionally. so, I can add a comment about prefix and easy to read